server/controllers/request.js

```diff
@@ -0,0 +1,77 @@
+// require Request Model
+const Request = {};
+
+exports.getRequestById = (req, res, next, id) => {
+  Request
+    .findById(id)
+    .exec((error, request) => {
+      if (error) {
+        return res.status(400).json({
+          error: "No request was found in DB"
+        })
+      }
+
+      req.request = request;
+      next();
+
+    })
+}
+
+exports.getAllRequest = (req, res) => {
+  let limit = req.query.limit ? parseInt(req.query.limit) : 8;
+  let sortBy = req.query.sortBy ? req.query.sortBy : "_id";
+
+  Request
+    .find()
+    .sort([[sortBy, 'ascending']])
+    .limit(limit)
+    .exec((error, requests) => {
+      if (error) {
+        return res.status(500).json({
+          error: "Error in Fetching the request from DB"
+        })
+      }
+
+      return res.status(200).json({ requests });
+
+    })
+}
+
+exports.createRequest = (req, res) => {
+  const request = new Request(req.body);
+  request.save((error, request) => {
+    if (error) {
+      return res.status(400).json({
+        error: "Unable to save request in DB"
+      })
+    }
+
+    return res.status(200).json({
+      request
+    })
+
+  })
+}
+
+exports.updateRequest = (req, res) => {

```

Comment:
This route should probably confirm that the logged in user is the one receiving the request

Reply:
I am sorry, I didn't understand. I did confirm by adding the `protect` middleware that the logged in user is receiving the request. Can you please clarify this for me ? If you meant by pull right request from the database, I think specifying the `_id` will be sufficient or we can also fetch the request from database using the `user_id`. Please, let me know about this.

Reply:
Never mind, I get it.